### ISSUE TITLE
Tpetra: importAndFillComplete fails for one-to-many distributions

### ISSUE THREAD

**Issue by kddevin** [trilinos/Trilinos#10008]
## Bug Report
@trilinos/tpetra @trilinos/zoltan2 @csiefer2 @tjfulle 

### Description
While debugging #10001, I'm trying to let Zoltan2 balance and migrate a matrix that is entirely on rank 0 initially; the resulting matrix should be spread across processors.

Below is a standalone code showing how I am trying to do the migration.   
@csiefer2  @tjfulle  Is my code using importAndFillComplete correctly?   
 It throws an error in lowCommunicationMakeColMapAndReindex, which is called from importAndFillComplete
```
 p=0: *** Caught standard std::exception of type 'std::invalid_argument' 
  /ascldap/users/kddevin/code/Trilinos.develop/packages/tpetra/core/src/Tpetra_Import_Util2.hpp:793:
  Throw number = 1
  Throw test that evaluated to true: PID == -1
  lowCommunicationMakeColMapAndReindex: Cannot figure out if PID is owned.
 [FAILED]  (0.0033 sec) CrsMatrix_double_int_longlong_Kokkos_Compat_KokkosSerialWrapperNode_Bug10001_UnitTest
```

Here's the code:
```  
  typedef Tpetra::global_size_t GST;
  const GST dummy = Teuchos::OrdinalTraits<GST>::invalid();

  typedef Tpetra::Map<LO,GO,NT> map_t;
  typedef Tpetra::Import<LO,GO,NT> import_t;
  typedef Tpetra::CrsMatrix<SC,LO,GO,NT> matrix_t;

  // Map with everything on rank zero
  const GO nRows = 25;
  const int nPerRow = 5;
  Teuchos::RCP<const map_t> mapOne =
           rcp(new map_t(dummy, (me == 0 ? nRows : 0), 0, comm));
    
  // Matrix using mapOne -- all data on rank 0
  matrix_t matrixOne(mapOne, nPerRow);

  if (me == 0) {
    Teuchos::Array<GO> gids(nPerRow);
    Teuchos::Array<SC> values(nPerRow);
    for (GO i = 0; i < nRows; i++) {
      // five-point stencil
      int nnz = 0;
      gids[nnz++] = i;
      if (((i-1) >= 0) && (((i-1)%nPerRow) < (i%nPerRow))) gids[nnz++] = i-1; // west nbor
      if (((i+1) < nRows) && (((i+1) % nPerRow) > (i%nPerRow))) gids[nnz++] = i+1; // east nbor
      if ((i-nPerRow) >= 0) gids[nnz++] = i-nPerRow; // south nbor  
      if ((i+nPerRow) < nRows) gids[nnz++] = i+nPerRow; // north nbor

      matrixOne.insertGlobalValues(i, gids(0, nnz), values(0, nnz));
    } 
  }
  matrixOne.fillComplete();

  // Import to matrix spread across all processors
  Teuchos::RCP<const map_t> mapMany = Teuchos::rcp (new map_t (nRows, 0, comm));

  import_t importer(mapOne, mapMany);

  Teuchos::RCP<matrix_t> matrixMany;
  matrixOne.importAndFillComplete(matrixMany, importer, mapMany, mapMany);
```

### Steps to Reproduce
Running with Serial node, SEMS modules, ascicgpu

**Comment by jhux2** [comment]
@kddevin Don't you want this instead?
```
matrixMany.importAndFillComplete(matrixOne, importer, mapMany, mapMany);
```

**Comment by kddevin** [comment]
Thanks, @jhux2  .  Not as I read the documentation; it says "import from this to the given destination matrix."  
https://docs.trilinos.org/dev/packages/tpetra/doc/html/classTpetra_1_1CrsMatrix.html#a759514cf7826750a3c3a3b318fe17cec
The documentation could never be wrong.  :)

**Comment by kddevin** [comment]
If, instead of having the entire matrix on rank 0 initially, I put the last row on rank 1, the code does what I want.  There must be something about rank 1 being empty in the first version.  ....

**Comment by csiefer2** [comment]
I suspect there's a subtle bug in the code because we never thought of this case.

**Comment by kddevin** [comment]
See reproducer in #10011

**Comment by tjfulle** [comment]
I'm building and running the reproducer in #10011 right now. Will update when more info is learned.

**Comment by tjfulle** [comment]
I believe I found the issue.  In [unpackAndCombineWithOwningPIDsCount ](https://github.com/trilinos/Trilinos/blob/fff6f703da27afbfb95dd50e01c21370b1270d24/packages/tpetra/core/src/Tpetra_Details_unpackCrsMatrixAndCombine_def.hpp#L1417) the `TargetPids` array will be set to -1 for locally owned entries.  This is interpreted by other procedures to mean "these PIDs are unknown".  After the call to `unpackAndCombineWithOwningPIDsCount` the -1 can be replaced with the actual PID of the calling (local) process.  #10011 has been updated to reflect this proposed change.
